### PR TITLE
Command support

### PR DESCRIPTION
**Pull request by wtk-evoirin**

### CODE REVIEW

File: apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h [hunk 1]
```diff
@@ -5,11 +5,26 @@
 extern "C" {
 #endif
 
+typedef struct iotc_command_queue_item_s  {
+    const char *command;
+    const char *ack_id;  
+} iotc_command_queue_item_t;
```
Comment (nik-markovic) [inline apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h:11]: Minor, Naming: iotc_command_queue_item_t should have camel case (per what our tpical project style does). Not sure if you are trying conform to the DA's project naming. If that's the case, then it's fine.

**Review by nik-markovic — APPROVED** [review]

**Review by nik-markovic — CHANGES_REQUESTED** [review]

File: apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h [hunk 2]
```diff
@@ -5,11 +5,26 @@
...
+
 extern void stop_iotconnect(void);
```
[the remark this answers is not in the document]
Comment (wtk-evoirin) [reply apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h:13]: Good point

File: apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.c [hunk 3]
```diff
@@ -116,6 +119,83 @@ static void on_connection_status(IotConnectMqttStatus status) {
...
+        iotc_command_queue_item_destroy(item);
+        return NULL;
+    }
+
+    return item;
+}
+
+static void on_command_receive(IotclC2dEventData data) {
+    const char                  *command        = iotcl_c2d_get_command(data);
+    const char                  *ack_id         = iotcl_c2d_get_ack_id(data);
```
[the remark this answers is not in the document]
Comment (wtk-evoirin) [reply apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.c:133]: This has been addressed.

File: apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.c [hunk 4]
```diff
@@ -116,6 +119,83 @@ static void on_connection_status(IotConnectMqttStatus status) {
...
+        IOTC_INFO("Command %s received with %s ACK ID\n", command, ack_id ? ack_id : "no");
+
+        queue_entry = iotc_command_queue_item_create(command, ack_id);
+
+        if (queue_entry == NULL) {
+            iotcl_mqtt_send_cmd_ack(ack_id, IOTCL_C2D_EVT_CMD_FAILED, "Internal error");
+            return;
+        }
+
+        if (pdTRUE == xQueueSendToBack(command_queue, &queue_entry, 0)) {
```
[the remark this answers is not in the document]
Comment (wtk-evoirin) [reply apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.c:152]: I have changed this.

File: apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h (hunk 1 above)
Comment (wtk-evoirin) [reply apps/common/examples/Network/IoTConnect_Client/src/iotc_app/iotc_app.h]: Yes, this is to stay consistent with the DA. I have removed the _t suffix.

**Review by nik-markovic — APPROVED** [review]

**Pull request merged**